### DISCUSSION TITLE
Problem with multiple alerts for a rule

### DISCUSSION THREAD

**Discussion by zshhh666666** [jertel/elastalert2#1503] [Q&A] [answered]
Whether elastalert2 can use multiple different types of alerts within the same rules?If so, how to configure it.

Here are my startup commands!

docker run --net=elastic -d --name elastalert --restart=always \
-v /opt/elk8.12.0/elastalert/alerttest/elastalert.yaml:/opt/elastalert/config.yaml \
-v /opt/elk8.12.0/elastalert/alerttest/rules:/opt/elastalert/rules \
-v /opt/elk8.12.0/elastalert/alerttest/certs:/opt/elastalert/certs \
-v /opt/elk8.12.0/elastalert/alerttest/smtpfile:/opt/elastalert/smtpfile \
-v /etc/localtime:/etc/localtime \
jertel/elastalert2:2.18.0 --verbose

I'm sure both methods work when used alone.I tried the following configuration,but only emails work.

![alert2](https://github.com/user-attachments/assets/9ea9eef9-f101-41dc-973d-4c8436865b2d)

I tried a different approach,The configuration error is displayed.

![alert1](https://github.com/user-attachments/assets/293753ed-3a61-44c1-a64b-d043143f06ed)

**Comment by jertel** [comment, marked as answer]
Yes. You are close. It should look like this:

```
...
alert:
- dingtalk
- email

dingtalk_something: somevalue
smtp_host: someHttpHost
...

```
